### PR TITLE
win_iis_webapppool: tests, check_mode, bugfixes

### PR DESCRIPTION
**Pull request by jborean93**
##### SUMMARY
Adds the following to win_iis_webapppool

* check mode support
* full attributes are now returned
* better idempotency checks
* bug fixes around casting of variables
* deprecated the string format for attributes in favour for a dict style parameter
* ability to set the enum or enum value for attributes
* integration tests
* updates on documentation
* probably more I can't think of

##### ISSUE TYPE
 - Feature Pull Request
 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME
win_iis_webapppool

##### ANSIBLE VERSION
```
ansible 2.4.0 (devel 8edcef7a84) last updated 2017/06/18 21:16:32 (GMT +1100)
  config file = None
  configured module search path = [u'/home/jordan/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /mnt/c/dev/linux-ansible/lib/ansible
  executable location = /mnt/c/dev/linux-ansible/bin/ansible
  python version = 2.7.13 (default, Jun 18 2017, 20:35:39) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION
This started to snowball into a bigger change as I was looking at the various issues and tried to get the tests to work.

Fixes
* https://github.com/ansible/ansible/issues/25705
* https://github.com/ansible/ansible/issues/25467
* https://github.com/ansible/ansible/issues/23951

### CODE REVIEW

File: lib/ansible/modules/windows/win_iis_webapppool.py [hunk 1]
```diff
@@ -27,125 +27,208 @@
...
+- name: restart an application pool (non-idempotent)
   win_iis_webapppool:
     name: AppPool
     state: restart
 
-- name: Changes application pool attributes without touching state
+- name: change application pool attributes using new dict style
+  win_iis_webapppool:
+    name: AppPool
+    attribute:
```
Comment (jhawkesworth) [inline lib/ansible/modules/windows/win_iis_webapppool.py]: Typo? attribute -> attributes?

Comment (jborean93) [reply lib/ansible/modules/windows/win_iis_webapppool.py]: Thanks will update

**Comment by pfar1** [comment]
I can confirm that i have now tested setting all properties via this module and all are now settable without issue.

For those interested, this is task i have, this has all the attributes specified:


```
- name: "Assuring {{iis_apppool_name}} Application pool"
  win_iis_webapppool:
   name: "{{iis_apppool_name}}"
   state: started
   attributes:
     name: "{{iis_apppool_name}}"
     managedRuntimeVersion: "{{iis_apppool_managedRuntimeVersion}}"
     enable32BitAppOnWin64: "{{iis_apppool_enable32BitAppOnWin64}}"
     managedPipelineMode: "{{iis_apppool_managedPipelineMode}}"
     startMode: "{{iis_apppool_startMode}}"
     queueLength: "{{iis_apppool_queueLength}}"
     cpu.limit: "{{iis_apppool_cpu_limit}}"
     cpu.action: "{{iis_apppool_cpu_action}}"
     cpu.smpAffinitized: "{{iis_apppool_cpu_smpAffinitized}}"
     cpu.smpProcessorAffinityMask: "{{iis_apppool_cpu_smpProcessorAffinityMask}}"
     cpu.smpProcessorAffinityMask2: "{{iis_apppool_cpu_smpProcessorAffinityMask2}}"
     cpu.processorGroup: "{{iis_apppool_cpu_processorGroup}}"
     cpu.numaNodeAssignment: "{{iis_apppool_cpu_numaNodeAssignment}}"
     cpu.numaNodeAffinityMode: "{{iis_apppool_cpu_numaNodeAffinityMode}}"
     processModel.identityType: "{{iis_apppool_processModel_identityType}}"
     processModel.loadUserProfile: "{{iis_apppool_processModel_loadUserProfile}}"
     processModel.idleTimeout: "{{iis_apppool_processModel_idleTimeout}}"
     processModel.idleTimeoutAction: "{{iis_apppool_processModel_idleTimeoutAction}}"
     processModel.pingingEnabled: "{{iis_apppool_processModel_pingingEnabled}}"
     processModel.pingResponseTime: "{{iis_apppool_processModel_pingResponseTime}}"
     processModel.pingInterval: "{{iis_apppool_processModel_pingInterval}}"
     processModel.shutdownTimeLimit: "{{iis_apppool_processModel_shutdownTimeLimit}}"
     processModel.startupTimeLimit: "{{iis_apppool_processModel_startupTimeLimit}}"
     processModel.maxProcesses: "{{iis_apppool_processModel_maxProcesses}}"
     failure.orphanWorkerProcess: "{{iis_apppool_failure_orphanWorkerProcess}}"
     failure.orphanActionExe: "{{iis_apppool_failure_orphanActionExe}}"
     failure.orphanActionParams: "{{iis_apppool_failure_orphanActionParams}}"
     failure.loadBalancerCapabilities: "{{iis_apppool_failure_loadBalancerCapabilities}}"
     failure.rapidFailProtection: "{{iis_apppool_failure_rapidFailProtection}}"
     failure.rapidFailProtectionInterval: "{{iis_apppool_failure_rapidFailProtectionInterval}}"
     failure.rapidFailProtectionMaxCrashes: "{{iis_apppool_failure_rapidFailProtectionMaxCrashes}}"
     failure.autoShutdownExe: "{{iis_apppool_failure_autoShutdownExe}}"
     failure.autoShutdownParams: "{{iis_apppool_failure_autoShutdownParams}}"
     recycling.disallowOverlappingRotation: "{{iis_apppool_recycling_disallowOverlappingRotation}}"
     recycling.disallowRotationOnConfigChange: "{{iis_apppool_recycling_disallowRotationOnConfigChange}}"
     recycling.periodicRestart.memory: "{{iis_apppool_recycling_periodicRestart_memory}}"
     recycling.periodicRestart.privateMemory: "{{iis_apppool_recycling_periodicRestart_privateMemory}}"
     recycling.periodicRestart.time: "{{iis_apppool_recycling_periodicRestart_time}}"
     recycling.periodicRestart.requests: "{{iis_apppool_recycling_periodicRestart_requests}}"

```

These are the variables i used at the time, please note, most of these are random settings due to testing the changes:

```
iis_apppool_managedRuntimeVersion: v4.0
iis_apppool_enable32BitAppOnWin64: false
iis_apppool_managedPipelineMode: Integrated
iis_apppool_startMode: AlwaysRunning
iis_apppool_queueLength: 1225
iis_apppool_cpu_limit: 0
iis_apppool_cpu_action: NoAction
iis_apppool_cpu_resetInterval: 5
iis_apppool_cpu_smpAffinitized: false
iis_apppool_cpu_smpProcessorAffinityMask: 4294967295
iis_apppool_cpu_smpProcessorAffinityMask2: 4294967295
iis_apppool_cpu_processorGroup: 0
iis_apppool_cpu_numaNodeAssignment: MostAvailableMemory
iis_apppool_cpu_numaNodeAffinityMode: Soft
iis_apppool_processModel_identityType: ApplicationPoolIdentity
iis_apppool_processModel_loadUserProfile: false
iis_apppool_processModel_idleTimeout: 0
iis_apppool_processModel_idleTimeoutAction: Terminate
iis_apppool_processModel_pingingEnabled: true
iis_apppool_processModel_pingResponseTime: "00:00:10"
iis_apppool_processModel_pingInterval: "00:00:10"
iis_apppool_processModel_startupTimeLimit: "00:00:10"
iis_apppool_processModel_shutdownTimeLimit: "00:02:30"
iis_apppool_processModel_maxProcesses: 2
iis_apppool_failure_orphanWorkerProcess: false
iis_apppool_failure_orphanActionExe:
iis_apppool_failure_orphanActionParams:
iis_apppool_failure_loadBalancerCapabilities: HttpLevel
iis_apppool_failure_rapidFailProtection: false
iis_apppool_failure_rapidFailProtectionInterval: "00:00:01:00"
iis_apppool_failure_rapidFailProtectionMaxCrashes: 20
iis_apppool_failure_autoShutdownExe:
iis_apppool_failure_autoShutdownParams:
iis_apppool_recycling_disallowOverlappingRotation: false
iis_apppool_recycling_disallowRotationOnConfigChange: false
iis_apppool_recycling_logEventOnRecycle: IsapiUnhealthy,OnDemand,Time,ConfigChange,PrivateMemory,Memory,Schedule,Requests
iis_apppool_recycling_periodicRestart_memory: 1024
iis_apppool_recycling_periodicRestart_privateMemory: 1024
iis_apppool_recycling_periodicRestart_time: "00:01:00"
iis_apppool_recycling_periodicRestart_requests: 100
```

**Comment by nitzmahone** [comment]
@pfar1 thanks for the extensive validation!

**Comment by jhawkesworth** [comment]
shipit

**Pull request merged**